docs/spec/dataclasses.rst

```diff
@@ -526,6 +532,56 @@ This includes, but is not limited to, the following semantics:
   dataclass must be explicitly annotated as e.g. ``x: ClassVar[Final[int]] =
   3``.
 
+Converters
+^^^^^^^^^^
+
+The ``converter`` parameter can be specified in a field definition to provide a single-argument callable
+used to convert values when assigning to the associated attribute. This feature allows for automatic type
+conversion and validation during object initialization and attribute assignment.
+
+Converter behavior:
+
+* For non-frozen dataclasses, the converter is used for all attribute assignment, including assignment
```

Comment:
I agree with rchen152 that this section is confusing. Why does the behavior for frozen classes need to be discussed at all here? A converter should work anywhere that assignment works — including initial assignment via the synthesized `__init__`. If a class is frozen, then subsequent assignments will generate an exception. That's how frozen classes work, so I don't think there's a need to explain that here.

Reply:
The intent was to ensure there's no ambiguity on frozen dataclasses, since one could mistakenly assume they don't "assign" their attributes.
I think here we'll just say "both frozen and non-frozen" and move on...